File: rust/src/models/batch_job_failures_response.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::batch_job_item_failure_response::BatchJobItemFailureResponse;

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct BatchJobFailuresResponse {
    pub data: Vec<BatchJobItemFailureResponse>,

    pub total_count: i32,
}

impl BatchJobFailuresResponse {
    pub fn new(data: Vec<BatchJobItemFailureResponse>, total_count: i32) -> Self {
        Self { data, total_count }
    }
}

File: typescript/src/models/batchJobFailuresResponse.ts
// this file is @generated
import {
  type BatchJobItemFailureResponse,
  BatchJobItemFailureResponseSerializer,
} from "./batchJobItemFailureResponse";

export interface BatchJobFailuresResponse {
  data: BatchJobItemFailureResponse[];

  totalCount: number;
}

export const BatchJobFailuresResponseSerializer = {
  _fromJsonObject(object: any): BatchJobFailuresResponse {
    return {
      data: object["data"].map((item: any) =>
        BatchJobItemFailureResponseSerializer._fromJsonObject(item)
      ),
      totalCount: object["total_count"],
    };
  },

  _toJsonObject(self: BatchJobFailuresResponse): any {
    return {
      data: self.data.map((item: any) =>
        BatchJobItemFailureResponseSerializer._toJsonObject(item)
      ),
      total_count: self.totalCount,
    };
  },
};
